Cargo.toml

```diff
@@ -21,6 +21,7 @@ members = [
     "crates/catalog/*",
     "crates/examples",
     "crates/iceberg",
+    "crates/integrations",
```

Comment:
Perhaps as a V2 of this integration it'd make sense to use features and/or move the datafusion dep down into the integrations crate (or maybe have a crate per integration).

To your point about polars or other extensions, if this concept grew the dependency load would get challenging for folks who don't need it.

Reply:
yes - I think this makes a lot of sense. I'd definitely go for this design. thanks

Reply:
@tshauck 
I updated the structure - to allow a specific crate per integration;
I'm not sure if I handled the feature-flags correctly - so PTAL and let me know what you think.

Reply:
Updating the integrations to be datafusion specific looks good to me. For the actual datafusion dependency, you could move that into the Cargo.toml of the new datafusion subcrate itself. So e.g...

```toml
[dependencies]
dashmap = { workspace = true }
datafusion = "37.0.0"
futures = { workspace = true }
iceberg = { workspace = true }
log = { workspace = true }
tokio = { workspace = true }
```

I think it's generally considered better to scope subcrate specific dependencies to the crate that needs it. I'm sorry for muddying the waters via the feature suggestion, but then I'm not sure the additional complexity of it would be needed.

Reply:
> I'm sorry for muddying the waters via the feature suggestion, but then I'm not sure the additional complexity of it would be needed.

Perhaps @liurenjie1024 @Xuanwo you can take another look - whats best practice here in terms of deps?